### ISSUE TITLE
Rexray does not start and obscure debuging message

### ISSUE THREAD

**Issue by davvdg** [codedellemc/rexray#564]
Hi, I'm trying to use rexray with no luck yet. Evently I'd like it to work with openstack but I can't even have it working in "dry mode", meaning out of the box: the daemon just won't start.

Here is my config: ubuntu 16.04, docker 1.12, rexray 0.3.3 or 0.5.1, no config file at all (so I assume, all working with default values).

When doing: sudo rexray start -l debug -f, I've got for 0.3.3 (I've trimmed the beginning...):

```
INFO[0000] initialized module instance                   address=unix:///run/docker/plugins/rexray.sock name=default-docker typeName=docker
INFO[0000] [linux]                                      
INFO[0000] [docker]                                     
INFO[0000] [libstorage]                                 
DEBU[0000] core get drivers                              osDrivers=[linux] storageDrivers=[libstorage] volumeDrivers=[docker]
INFO[0000] docker volume driver initialized              availabilityZone= iops= moduleName=default-docker provider=docker size= volumeRootPath=/data volumeType=
INFO[0000] os driver initialized                         moduleName=default-docker provider=linux
DEBU[0000] checking volume path cache setting            pathCache=true
INFO[0000] vdm.List                                      driverName=docker moduleName=default-docker
INFO[0000] listing volumes                               driverName=docker moduleName=default-docker
ERRO[0000] default module(s) failed to start            
INFO[0000] service sent registered modules start signals 
ERRO[0000] service initialized failed                   
INFO[0000] error initializing driver                     m=&{r:0xc820304540 name:default-docker addr:unix:///run/docker/plugins/rexray.sock desc:The default docker module.} m.r=&{Config:0xc820301620 OS:0xc82032f9f0 Volume:0xc820301ec0 Storage:0xc82032fa00 Context:default-docker drivers:map[XtremIO:0xc820304600 Rackspace:0xc8202fcd40 Openstack:0xc8203045a0 ScaleIO:0xc820302420 docker:0xc820026300 virtualbox:0xc8203017e0 VMAX:0xc8202b9e50 linux:0xc8200262f8 ec2:0xc8203021b0 gce:0xc8202fc880 Isilon:0xc820306800]}
```

What is this last one line telling me about the bug ?

for 0.5.1:

```
INFO[0000] configured logging                            libstorage.logging.level=debug server=honeysuckle-buffalo-kw time=1474552510858
INFO[0000] initializing server                           server=honeysuckle-buffalo-kw time=1474552510858
INFO[0000] initializing default endpoint                 autoEndpointMode=unix server=honeysuckle-buffalo-kw time=1474552510859
INFO[0000] initializing auto unix endpoint               endpoint=libstorage.server.endpoints.localhost server=honeysuckle-buffalo-kw time=1474552510859
DEBU[0000] endpoint info                                 address=unix:///var/run/libstorage/667611626.sock endpoint=libstorage.server.endpoints.localhost server=honeysuckle-buffalo-kw time=1474552510860
DEBU[0000] created scoped scope                          new=libstorage.server.endpoints.localhost parentScopes=libstorage.server,rexray,
INFO[0000] configured endpoint                           address=unix:///var/run/libstorage/667611626.sock endpoint=localhost server=honeysuckle-buffalo-kw time=1474552510861
INFO[0000] server created                                server=honeysuckle-buffalo-kw time=1474552510862
INFO[0000] initialized endpoints                         server=honeysuckle-buffalo-kw time=1474552510862
INFO[0000] initializing server services                  server=honeysuckle-buffalo-kw time=1474552510862
DEBU[0000] configured result schema validation           enabled=false server=honeysuckle-buffalo-kw time=1474552510862
ERRO[0000] default module(s) failed to initialize        error.configKey=libstorage.server.services error.obj=<nil> time=1474552510863
ERRO[0000] error starting rex-ray                        error.configKey=libstorage.server.services error.obj=<nil> time=1474552510863
DEBU[0000] completed cli execution                       time=1474552510863
INFO[0000] exiting process                               time=1474552510863
DEBU[0000] completed onExit at end of program            time=1474552510863
```

Thank you for your help, folks !

David

**Comment by kacole2** [comment]
@davvdg what storage platform are you testing with? You have to use a configuration file no matter what. I would suggest to start out trying with VirtualBox

**Comment by kacole2** [comment]
@davvdg The log is saying that it's looking at a configuration file ( which isn't there )and trying to initialize a driver for a specific storage platform.

**Comment by davvdg** [comment]
My vm are run by Openstack, the storage is cinder.

2016-09-22 16:29 GMT+02:00 Kendrick Coleman <notifications@github.com>:

**Comment by kacole2** [comment]
@davvdg the OpenStack driver is in v0.3.3 right now. It will be added to v0.5 soon (https://github.com/emccode/libstorage/pull/243)

Step 1. [Install](http://rexray.readthedocs.io/en/latest/user-guide/installation/#rex-ray-033)
Step 2. [Create a configuration file using OpenStack](http://rexray.readthedocs.io/en/v0.3.3/user-guide/storage-providers/#openstack)
Step 3. Place the configuration file at `/etc/rexray/config.yml`
Step 4. Start with `rexray start`

**Comment by davvdg** [comment]
Hi. I've just tested again with my openstack config file, for 0.3.3 and adapted for 0.5.1, but still no luck.

For 0.3.3:
/etc/rexray/config.yml
```
rexray:
  loglevel: debug
  storageDrivers:
  - openstack
openstack:
  authUrl:              https://xxxxxxxxxx:5000/v3
  username:             xxxxxx
  password:             xxxxxxx
  tenantId:             xxxx0x0x0x0x
  tenantName:           xxx
  projectName:          xxx
  identityApiVersion:   3
  userDomainName:       xxxxxxx
  projectDomainName:    default
  cacert:               /opt/openstack/certs/xxxxxx.pem
  availabilityZoneName: nova
  regionName:           region-a.geo-1
  domainId:             0
  domainName:           xxxxxxx
  userId:               0
```
Error for 0.3.3:
```
DEBU[0000] error initializing driver                     driverName=Openstack error.provider=Openstack error.identityEndpoint="https://xxxxxx:5000/v3" error.tenantName=xxx error.userId=0 error.domainName=xxxx error.availabilityZone=nova error.instanceId=xxxxxx error.tenantId=xxxxx error.region=region-a.geo-1 error.domainId=0 error.moduleName=default-docker error.userName=xxxxx error.password="******"
DEBU[0000] checking volume path cache setting            pathCache=true
INFO[0000] vdm.List                                      driverName=docker moduleName=default-docker
INFO[0000] listing volumes                               driverName=docker moduleName=default-docker
ERRO[0000] default module(s) failed to start            
INFO[0000] service sent registered modules start signals 
ERRO[0000] service initialized failed                   
INFO[0000] error initializing driver                     m=&{r:0xc82030d080 name:default-docker addr:unix:///run/docker/plugins/rexray.sock desc:The default docker module.} m.r=&{Config:0xc82031c7a0 OS:0xc8203e6b80 Volume:0xc8203bec20 Storage:0xc8203e6b90 Context:default-docker drivers:map[ec2:0xc820310ed0 Isilon:0xc82031b800 docker:0xc820026380 linux:0xc820026388 VMAX:0xc820316050 XtremIO:0xc82030d0e0 ScaleIO:0xc820311140 virtualbox:0xc82031c9e0 gce:0xc82030b600 Openstack:0xc82030d140 Rackspace:0xc82030b340]}
```

Now for 0.5.1:
/etc/rexray/config.yml
```
rexray:
  loglevel: debug
libstorage:
  service: openstack
openstack:
  authUrl:              https://xxxxxxxxxx:5000/v3
  username:             xxxxxx
  password:             xxxxxxx
  tenantId:             xxxx0x0x0x0x
  tenantName:           xxx
  projectName:          xxx
  identityApiVersion:   3
  userDomainName:       xxxxxxx
  projectDomainName:    default
  cacert:               /opt/openstack/certs/xxxxxx.pem
  availabilityZoneName: nova
  regionName:           region-a.geo-1
  domainId:             0
  domainName:           xxxxxxx
  userId:               0
```
and error, for 0.5.1
```
INFO[0000] configured logging                            libstorage.logging.level=debug server=cookie-jester-qa time=1474555448061
INFO[0000] initializing server                           server=cookie-jester-qa time=1474555448062
INFO[0000] initializing default endpoint                 autoEndpointMode=unix server=cookie-jester-qa time=1474555448063
INFO[0000] initializing auto unix endpoint               endpoint=libstorage.server.endpoints.localhost server=cookie-jester-qa time=1474555448064
DEBU[0000] endpoint info                                 address=unix:///var/run/libstorage/648800199.sock endpoint=libstorage.server.endpoints.localhost server=cookie-jester-qa time=1474555448064
DEBU[0000] created scoped scope                          new=libstorage.server.endpoints.localhost parentScopes=libstorage.server,rexray,
INFO[0000] configured endpoint                           address=unix:///var/run/libstorage/648800199.sock endpoint=localhost server=cookie-jester-qa time=1474555448065
INFO[0000] server created                                server=cookie-jester-qa time=1474555448065
INFO[0000] initialized endpoints                         server=cookie-jester-qa time=1474555448065
INFO[0000] initializing server services                  server=cookie-jester-qa time=1474555448065
DEBU[0000] configured result schema validation           enabled=false server=cookie-jester-qa time=1474555448066
DEBU[0000] got services map                              count=1 server=cookie-jester-qa time=1474555448066
DEBU[0000] processing service config                     server=cookie-jester-qa service=openstack time=1474555448066
DEBU[0000] getting scoped config for service             scope=libstorage.server.services.openstack server=cookie-jester-qa service=openstack time=1474555448066
DEBU[0000] created scoped scope                          new=libstorage.server.services.openstack parentScopes=libstorage.server,rexray,
DEBU[0000] got driver name                               driverName=openstack server=cookie-jester-qa service=openstack time=1474555448067
ERRO[0000] default module(s) failed to initialize        error.driver=openstack time=1474555448067
ERRO[0000] error starting rex-ray                        error.driver=openstack time=1474555448067
DEBU[0000] completed cli execution                       time=1474555448067
INFO[0000] exiting process                               time=1474555448068
DEBU[0000] completed onExit at end of program            time=147455544806

```

**Comment by davvdg** [comment]
I should add that when I use openstack CLI, I don't need to provide the following env variables:

- availabilityZoneName
- regionName
- domainId
- domainName
- userId

but rexray was not happy with them missing so I added them with some dummy values...

**Comment by kacole2** [comment]
@davvdg is it working? I don't have any experience on the OpenStack driver but all the configuration variables will be [here](http://rexray.readthedocs.io/en/v0.3.3/user-guide/storage-providers/#openstack)

**Comment by davvdg** [comment]
nope. I still have the same error:
```
INFO[0000] error initializing driver                     m=&{r:0xc820319080
name:default-docker addr:unix:///run/docker/plugins/rexray.sock desc:The
default docker module.} m.r=&{Config:0xc820324800 OS:0xc8203f0d20
Volume:0xc8203beca0 Storage:0xc8203f0d30 Context:default-docker
drivers:map[ScaleIO:0xc820314db0 XtremIO:0xc8203190e0 docker:0xc820026378
Rackspace:0xc820311400 linux:0xc820026380 gce:0xc8203112c0
Isilon:0xc820323910 ec2:0xc8203151d0 Openstack:0xc820319140
virtualbox:0xc820324ac0 VMAX:0xc82031a370]}
```

I think it is not related to openstack, it is related to the default-docker
that failed to initialized. I still have this message even without
configuring openstack at all.

2016-09-22 17:08 GMT+02:00 Kendrick Coleman <notifications@github.com>:

**Comment by MatMaul** [comment]
you have better logs when trying to use rexray CLI directly (without the Docker integration), using "rexray volume ls" for example.
For the variables, availabilityZoneName and regionName are optional (at least in the last version of the driver available in https://github.com/emccode/libstorage/pull/243, I don't know for v0.3.3).
For the domain and user name and/or id it mainly depends of the keystone version you are using.

The openstack driver is NOT available in the current rexray 0.5.x, so if you don't do a custom build with my PR you can skip the part where you try with 0.5.x, it will always fail :)
